### ISSUE TITLE
Issue when running behind Nginx Reverse Proxy

### ISSUE THREAD

**Issue by ricardofalcao** [dzikoysk/reposilite#95]
Hello, I was trying the repository manager and for the most part everything seems pretty fine, but when I tried using with the Nginx reverse proxy everything worked but the files could not be retrieved. Seems like the server interpreted the request as a folder instead of giving me the raw file:
![Screenshot_20200601_012050](https://user-images.githubusercontent.com/23485259/83366318-25a9b900-a3a6-11ea-8f65-63e6448f8eb1.png)

My current nginx settings are the following:
```
location / {
         proxy_set_header        X-Forwarded-For $proxy_add_x_forwarded_for;
         proxy_set_header Host $host;
         proxy_set_header X-Real-IP  $remote_addr;
         proxy_pass              http://localhost:9090/;
}
```

Thanks again for the project, and I will try to investigate the issue myself too.

**Comment by dzikoysk** [comment]
Hey, thanks 👋 

Reposilite also displays this page if the requested file is not found. The error might be a result of modified uri, could you make a request to this metadata file, but with `/api` prefix? `localhost/api/{metadata path}` :)

**Comment by ricardofalcao** [comment]
Hey, it returns the following:
```json
{
   "type":"file",
   "name":"maven-metadata.xml",
   "contentLength":609
}
```

**Comment by dzikoysk** [comment]
Could you display the value of `window.REPOSILITE_MESSAGE` using the  DevTools console?

**Comment by ricardofalcao** [comment]
```
"Authorization credentials are not specified"
```

It seems that I forgot that I had enabled authentication, so it makes sense that I could not access it through the web interface, although I can list every folder and repository. But when I try to compile a maven project with a dependency from the repository (with authentication set up in the settings.xml), it says that the artifact was not found:
```
Could not resolve dependencies for project org.github.paperspigot:paperspigot-api:jar:1.8.8-R0.1-SNAPSHOT: Could not find artifact io.github.waterfallmc:olimpo-chat:jar:1.14-SNAPSHOT in cubeland-repo
```

**Comment by dzikoysk** [comment]
The security issue related to the lack of auth process for `/api` requests is already resolved and will be available in 2.5.1

Are you sure that the `olimpo-chat:jar:1.14-SNAPSHOT` builds located in `snapshots` repository contains required by Maven protocol information? Here is an example: https://repo.panda-lang.org/snapshots/org/panda-lang/nanomaven/2.1.2-SNAPSHOT

**Comment by ricardofalcao** [comment]
Great to hear. Although I find that it is not the right way to use it, I usually also put my snapshot version on the releases repository, because it is a private repository. Still, I tried to move everything to the snapshots repository, but nothing changed, the artifact still could not be found. I also tried to disable the authentication, but also nothing changed. I have double-checked that the files are present in the repository.
![Screenshot_20200601_111021](https://user-images.githubusercontent.com/23485259/83399078-7d711000-a3f8-11ea-9c3f-ee7d2cae23be.png)

**Comment by dzikoysk** [comment]
Could you run your maven build with enabled debug (`-X`). You should be able to see full response from reposilite.

**Comment by ricardofalcao** [comment]
I have attached the output log:
[output.txt](https://github.com/dzikoysk/reposilite/files/4710866/output.txt)

**Comment by dzikoysk** [comment]
Could you try 2.5.1? https://repo.panda-lang.org/org/panda-lang/reposilite/2.5.1/reposilite-2.5.1.jar
You should be able to see some errors during the build process

**Comment by ricardofalcao** [comment]
I have uploaded the log: https://pastebin.com/kMi7GQzZ
It seems to not be able to find the maven metadata file for the olimpo-parent artifact altough in the picture below it clearly exists the specified file:

![Screenshot_20200601_145523](https://user-images.githubusercontent.com/23485259/83416012-f03db380-a417-11ea-8330-8c418ad5b93f.png)

**Comment by dzikoysk** [comment]
Reposilite does not use local metadata files anyway, so it doesn't matter.

Ok, well... the problem is that because of the `-SNAPSHOT` modifier, Maven requests `groupId/artifactId/version/maven-metadata.xml` file which is unsupported for standard repositories. It's not a bug, because Maven specification does not provide the possibility of snapshot releases in standard repositories. I've marked it as an enhancement request.

I'll probably implement it in 2-4 days, because I'm a little bussy at this moment.

**Comment by ricardofalcao** [comment]
Thank you very much, take your time!

**Comment by dzikoysk** [comment]
Could you try to use https://repo.panda-lang.org/org/panda-lang/reposilite/2.5.2/reposilite-2.5.2.jar? It generates fake metadata file for this kind of requests, maybe that's enough.

**Comment by ricardofalcao** [comment]
It seems to be working, great!  Although when my project fetches the dependencies it gives a warning saying that Checksum validation failed. That might not be related to the recent changes because I seem to remember seeing those warnings in the 2.5.0 release. Is this a known issue?

**Comment by dzikoysk** [comment]
Checksum validation always fails, because Reposilite doesn't trust the metadata files offered by client - its generated by server, based on the repository content to ensure consistency. It's fine and it does not affect your builds.

**Comment by dzikoysk** [comment]
I've released official 2.5.2, it contains some changes comparing to the version above.

https://github.com/dzikoysk/reposilite/releases
